src/common/macro/src/row/utils.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright 2023 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use greptime_proto::v1::ColumnDataType;
+use once_cell::sync::Lazy;
+use quote::format_ident;
+use syn::{
+    AngleBracketedGenericArguments, Data, DataStruct, Fields, FieldsNamed, GenericArgument, Ident,
+    Path, PathArguments, PathSegment, Result, Type, TypePath, TypeReference,
+};
+
+use crate::row::attribute::{find_column_attribute, parse_column_attribute, ColumnAttribute};
+
+static SEMANTIC_TYPES: Lazy<HashMap<&'static str, SemanticType>> = Lazy::new(|| {
+    HashMap::from([
+        ("field", SemanticType::Field),
+        ("tag", SemanticType::Tag),
+        ("timestamp", SemanticType::Timestamp),
+    ])
+});
+
+static DATATYPE_TO_COLUMN_DATA_TYPE: Lazy<HashMap<&'static str, ColumnDataType>> =
+    Lazy::new(|| {
+        HashMap::from([
+            // Timestamp
+            ("timestampsecond", ColumnDataType::TimestampSecond),
+            ("timestampmillisecond", ColumnDataType::TimestampMillisecond),
+            (
+                "timestamptimemicrosecond",
```

Comment:
typo?

Reply:
resolved in [23ade5d](https://github.com/GreptimeTeam/greptimedb/pull/6778/commits/23ade5d144a49809751416b2c1a71d115d68c4fb)